text/src/autogluon/text/automm/utils.py

```diff
@@ -503,6 +508,72 @@ def create_model(
     else:
         raise ValueError(f"No available models for {names}")
 
+def save_pretrained_model(
+    model: ModuleList,
+    config: DictConfig,
+    path: str
+) -> DictConfig:
+    '''
+    Saving the pretrained weights for offline deployment. Used by turing "standalone=True" in predictor. 
+
+    Parameters
+    ----------
+    model
+        A modulelist containing all models.
+    config
+        A DictConfig object. The model config should be accessible by "config.model".
+    path
+        The saving path to the pretained weights i.e. config.json for "clip" and "hf_text", 'pkl' for "timm_image"
+    '''
+    for idx, model_name in enumerate(config.model.names):
+        if model_name == "clip" or "hf_text" in model_name:
+            model[idx].model.save_pretrained(os.path.join(path,model_name))
+            model_config = getattr(config.model, model_name)
+            model_config.checkpoint_name = os.path.join('local://',model_name)
+        if model_name == "timm_image":
+            model_config = getattr(config.model, model_name)
+            if not os.path.exists(os.path.join(path,model_name)):
+                os.makedirs(os.path.join(path,model_name))
+            torch.save(
+                model[idx].model.state_dict(),
+                os.path.join(path,model_name,model_config.checkpoint_name+'.pkl')
```

Comment:
It's unclear why do we save a pretrained timm checkpoint in a 'pkl' file.

Reply:
I think we are safe to remove the check here.

Reply:
Can we save to `.pt` ?

Reply:
timm uses `.pth`.

Reply:
Great! Thanks for the advice. It is suggested no formatting difference between `.pkl`, `.pt` and `.pth`, except for a difference in suffix, eg [https://blog.karatos.in/a?ID=01600-da733521-cc0a-4be1-a696-41930451a6a9](url). We are going to use `.pth` to get aligned with timm.

Reply:
This block is now removed in [https://github.com/awslabs/autogluon/pull/1575/commits/b02bebb383d4e004d00380814baef9293977aea4](url)